public/main.js

```diff
@@ -0,0 +1,404 @@
+			.selectAll('line')
+			.data(xScale.ticks(20))
+			.enter()
+			.append('line')
+				.attr('x1', xScale)
+				.attr('x2', xScale)
+				.attr('y1', 30)
+				.attr('y2', canvas_height-25)
+				.style('stroke', '#ccc');
+
+
+
+
+		var todayline = svg.append('line')
+			.datum(today)
+			// .datum(new Date(2014,1,14))
+			.attr('x1', xScale)
+			.attr('x2', xScale)
+			.attr('y1', 0)
+			.attr('y2', canvas_height - 25)
+			.style('stroke', '#c00');
+
+		d3.select('#chart-canvas').style('height', canvas_height + 'px');
+
+
+		ganttBarContainer = d3.select('#gantt-bar-container')
+			.on('mousemove', function(d, i) {
+				// Place mouse move on bar-container so the tooltip renders over the bars but sets to the xy of the bar it tips
+				var xy = d3.mouse(this);
+				// Update Tooltip Position & value
+				tooltip
+					.style('left', xy[0] + 'px')
+					.style('top', xy[1] + 'px');
+			});
+
+		ganttBarContainer.append('div')
+			.datum(today)
+			.text('Today')
+			.attr('class', 'todaymarker')
+			.style('position', 'absolute')
+			// height of "today" slightly higher so as not to interfere with milestones
+			.style('top', '-50px')
+			.style('left', function(d) { return (xScale(d) -1) + 'px' });
+
+	}
+
+	var tooltip = d3.select('#tooltip');
+
+	function render() {
+
+		var filteredData = data;
+
+		if (filter_selector) {
+			if (filter_selector[1] == "All") {
+				filteredData = data.filter(function(d) { return d});
+			} else if (filter_selector[0] == "label" ) {
+				filteredData = data.filter(function(d) {
+					if (d.label) { 
+						var arr = d.label.indexOf(filter_selector[1]);
+						if (arr > -1) {
+							return d;
+						}
+					}
+				});
+			} else if (filter_selector[0] == "assignee" ) {
+				filteredData = data.filter(function(d) {
+					if (d.assignee) { 
+						var arr = d.assignee.indexOf(filter_selector[1]);
+						if (arr > -1) {
+							return d;
+						}
+					}
+				});
+			} else {
+				filteredData = data.filter(function(d) { return d[filter_selector[0]] == filter_selector[1]});	
+			}
+		}
+
+ 		var barWrappers = ganttBarContainer.selectAll('.bar-wrapper')
+			.data(filteredData, function(d,i) { return d.id});
+
+ 		var bwe = barWrappers
+			.enter()
+			.append('div')
+			.attr('class', function(d) { return 'bar-wrapper '+ d.assignee})
+			.on('mouseover', function(d, i) {
+				var tt = '';
+				tt += '<p class="heading"><span id="keyword"><b>' + d.title+ '</b></span></p>';
+				tt += '<p class="indent"><span id="bar-data">' + d.assignee + '</span></p>';
+				//tt += '<p class="indent"><span id="bar-data">' + d.priority + " hours" + '</span></p>';
+				//tt += '<p class="indent"><span id="cpcVal">' + dateFormat(d["created_at"]) + ' - ' + dateFormat(d["milestone"]["due_on"]) + '</span></p>';
+				tt += '<p class="indent"><span id="cpcVal">' + 'Start date: ' + titleFormat(d.created_at) + '</span></p>';
+				tt += '<p class="indent"><span id="cpcVal">' + 'Due date: ' + titleFormat(d.due_on) + '</span></p>';
```

Comment:
Please update the above code to use the JQuery API for creating elements, and have the text from github added in using the safe `.text()` method (otherwise we're inviting an XSS vulnerability)